src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -60,6 +60,28 @@ def discardConflictingDocument(couchDbInstance, data, result):
         return result
 
 
+def shrinkLargeFJR(couchDbInstance, sizeLimit=8e6):
+    """
+    Look at the CouchDB database queue and empty documents
+    with a 'fwjr' field larger than sizeLimit.
+    :param couchDbInstance: couchdb database instance
+    :param sizeLimit: integer with the limit number of bytes
+    :return: None
+    """
+    for doc in couchDbInstance._queue:
+        if sys.getsizeof(json.dumps(doc)) > sizeLimit:
+            if 'fwjr' in doc:
+                errMsg = f"The 'fwjr' attribute for jobid: {doc.get('jobid')} will "
+                errMsg += f"be empty because it is too large."
+                logging.warning(errMsg)
+                doc['fwjr'] = dict()
```

Comment:
That was my original suggestion too, why not to apply gzip to FWJR part entirely for all docs and only keep meta-data in unzipped form. Doing this way we get another benefit that size of the database will be much smaller, all docs will be in the same format, e.g. meta-data used by views are in unzipped form, and actual meta-data of FWJR will be zipped. On a client side whoever needs to use FWJR part will unzip it, etc.

Reply:
I am not sure who actually consumes this `fwjr` content and whether it is even consumed by any of the couchdb views. IF it is, then we cannot have compressed data stored in CouchDB.

I'd suggest to investigate and make this change through a new GH issue/PR, instead of refactoring things in here.

Reply:
I understand your point and I agree that nothing guarantees that that document will be smaller than 8MB.
Looking into the `recordInCouch` method of the ChangeState module, my main concern goes towards the `mask` attribute.

Nonetheless, we do not have enough information and know how to properly deal with that document in case of other fields are still larger than the configured document size limit in CouchDB. Having said that, I'd rather see an explicit error/exception with that component, such that we are able to properly debug it and come up with the best way to address it.